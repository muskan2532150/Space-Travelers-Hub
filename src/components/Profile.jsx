import React, { useState } from 'react';
import { useSelector } from 'react-redux';

const Profile = () => {
  const { mission, rockets } = useSelector((state) => state);

  const allMissions = mission.data.filter((msn) => msn.mission);
  const bookedRockets = rockets.rockets.filter((rkt) => rkt.reserved);

  const missionElements = (
    <ul>
      {allMissions.map((msn) => (
        <li key={msn.mission_id}>{msn.mission_name}</li>
      ))}
    </ul>
  );
  const rocketElements = (
    <ul>
      {bookedRockets.map((rkt) => (
        <li key={rkt.id}>{rkt.rocket_name}</li>
      ))}
    </ul>
  );

  return (
    <div className="profile">
      <div className="profile-missions">
        <h2 className="profile-rocket-missions">My Missions</h2>
        {missionElements}
      </div>
      <div className="profile-rockets">
        <h2 className="profile-rocket-header">My Rockets</h2>
        {rocketElements}
      </div>
    </div>
  );
};

export default Profile;
