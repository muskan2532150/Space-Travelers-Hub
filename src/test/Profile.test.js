import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../redux/Store";
import Profile from "../components/Profile";

test('Demo test', () => {
    expect((1 + 2)).toBe(3);
});


test('renders Profile Component', () => {
    const profile = render(<Provider store={store}><Profile/></Provider>);
    expect(profile).toMatchSnapshot();
});

