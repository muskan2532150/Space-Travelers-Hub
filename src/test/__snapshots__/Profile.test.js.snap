// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile"
      >
        <div
          class="profile-missions"
        >
          <h2
            class="profile-rocket-missions"
          >
            My Missions
          </h2>
          <ul>
            <li>
              No Joined Mission 
            </li>
          </ul>
        </div>
        <div
          class="profile-rockets"
        >
          <h2
            class="profile-rocket-header"
          >
            My Rockets
          </h2>
          <ul>
            <li>
              No Rocket Reseverd
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile"
    >
      <div
        class="profile-missions"
      >
        <h2
          class="profile-rocket-missions"
        >
          My Missions
        </h2>
        <ul>
          <li>
            No Joined Mission 
          </li>
        </ul>
      </div>
      <div
        class="profile-rockets"
      >
        <h2
          class="profile-rocket-header"
        >
          My Rockets
        </h2>
        <ul>
          <li>
            No Rocket Reseverd
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
